netneurotools/stats.py

```diff
@@ -784,3 +785,148 @@ def gen_spinsamples(coords, hemiid, n_rotate=1000, check_duplicates=True,
+        Whether to use adjusted r squares. Default: True
+    verbose : bool, optional
+        Whether to print debug messages. Default: False
+
+    Returns
+    -------
+    model_metrics : dict
+        The dominance metrics, currently containing `individual_dominance`,
+        `partial_dominance`, `total_dominance`, and `full_r_sq`.
+    model_r_sq : dict
+        Contains all model r squares
+
+    Notes
+    -----
+    Example usage
+
+    .. code:: python
+
+        from netneurotools.stats import get_dominance_stats
+        from sklearn.datasets import load_boston
+        X, y = load_boston(return_X_y=True)
+        model_metrics, model_r_sq = get_dominance_stats(X, y)
+
+    To compare with [DA1]_, use `use_adjusted_r_sq=False`
+
+    .. code:: python
+
+        from dominance_analysis import Dominance_Datasets
+        from dominance_analysis import Dominance
+        boston_dataset=Dominance_Datasets.get_boston()
+        dominance_regression=Dominance(data=boston_dataset,
+                                       target='House_Price',objective=1)
+        incr_variable_rsquare=dominance_regression.incremental_rsquare()
+        dominance_regression.dominance_stats()
+
+    References
+    ----------
+    .. [DA1] https://github.com/dominance-analysis/dominance-analysis
+
+    """
+
+    from itertools import combinations
+    from sklearn.linear_model import LinearRegression
+
+    # sklearn linear regression wrapper
+    def get_reg_r_sq(X, y):
+        lin_reg = LinearRegression()
+        lin_reg.fit(X, y)
+        yhat = lin_reg.predict(X)
+        SS_Residual = sum((y - yhat) ** 2)
+        SS_Total = sum((y - np.mean(y)) ** 2)
+        r_squared = 1 - (float(SS_Residual)) / SS_Total
+        adjusted_r_squared = 1 - (1 - r_squared) * \
+            (len(y) - 1) / (len(y) - X.shape[1] - 1)
+        if use_adjusted_r_sq:
+            return adjusted_r_squared
+        else:
+            return r_squared
+
+    def complete_model_rsquare(X, y):
+        return get_reg_r_sq(X, y)
+
+    # generate all predictor combinations in list (num of predictors) of lists
+    n_predictor = X.shape[-1]
+    # n_comb_len_group = n_predictor - 1
+    predictor_combs = [list(combinations(range(n_predictor), i))
+                       for i in range(1, n_predictor + 1)]
+    if verbose:
+        print(f"[Dominance analysis] Generated \
+              {len([_ for i in predictor_combs for _ in i])} combinations")
+
+    # get all r_sq's
+    model_r_sq = dict([])
+    for len_group in tqdm(predictor_combs, desc='num-of-predictor loop',
+                          disable=~verbose):
+        for idx_tuple in tqdm(len_group, desc='insider loop',
+                              disable=~verbose):
+            r_sq = get_reg_r_sq(X[:, idx_tuple], y)
+            model_r_sq[idx_tuple] = r_sq
+    if verbose:
+        print(f"[Dominance analysis] Acquired {len(model_r_sq)} r^2's")
+
+    # getting all model metrics
+    model_metrics = dict([])
```

Comment:
```suggestion
    model_metrics = dict()
```

Why the empty list here? I don't think the `dict` constructor will do anything with this, so this could just be `dict()`.